### PR TITLE
[bitwarden] Fix invalid session token retry loop

### PR DESCRIPTION
**Pull request by clins1994**
## Description

Related issues: #26943, #26173, #26574

### Bug Fix
- Fix infinite retry loop when unlocking vault fails with an invalid session token. The previous retry logic had no recursion guard, causing an endless "Unlocking vault..." / "Clearing session and retrying..." loop.
- Perform a full `bw logout` before retrying, ensuring the CLI starts from a clean state instead of only clearing the in-memory session token.
- Keep loading state active during retry so the UI does not flash back to the login form.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by clins1994** [comment]
@greptileai there were unrelated changes in the pr. I removed them. follow-up review please

**Comment by clins1994** [comment]
@greptileai fresh review please 🤓

**Comment by clins1994** [comment]
if the custom loading screen feels like too much, feel free to replace with a generic loading screen 👀 

thought it'd be fun 🤡 🎉

File: extensions/bitwarden/src/components/UnlockForm.tsx [hunk 1]
```diff
@@ -112,6 +127,15 @@ const UnlockForm = ({ pendingAction = Promise.resolve() }: UnlockFormProps) => {
...
 
+  if (isLoading) {
+    const unlockGif = path.join(environment.assetsPath, "unlock.gif");
+    return (
+      <List>
+        <List.EmptyView icon={unlockGif} title="" />
+      </List>
+    );
+  }
+
```
Comment (jomifepe) [inline extensions/bitwarden/src/components/UnlockForm.tsx]: I appreciate the effort, but I don't think it fits the theme 😄 Let's leave it as it was, loading is already indicated by the list.

File: extensions/bitwarden/src/components/UnlockForm.tsx [hunk 2]
```diff
@@ -60,8 +74,9 @@ const UnlockForm = ({ pendingAction = Promise.resolve() }: UnlockFormProps) => {
       toast.title = "Unlocking vault...";
       const { error: unlockError } = await bitwarden.unlock(password);
       if (unlockError) {
-        if (unlockError instanceof InvalidSessionTokenError) {
-          return onSubmit({ retryInvalidSessionToken: true });
+        if (unlockError instanceof InvalidSessionTokenError && !args?.retryInvalidSessionToken) {
+          await bitwarden.logout({ reason: "Invalid session token" });
```
Comment (jomifepe) [inline extensions/bitwarden/src/components/UnlockForm.tsx:66]: @0xdhrv I noticed you [removed the logout call](https://github.com/raycast/extensions/pull/26856/changes/b9102c45310c3fc710761bbf4b23713f366f27d9#diff-7ca6a791257e647a332d84d73c7bc7b77f13155c8fb5f9f3c626edd639f44666L775) before merging https://github.com/raycast/extensions/pull/26856. 

Did you find any issue with it? I don't think firing the listeners could cause an issue. 

Not trying to blame, genuinely asking.

---

That said, I think I would prefer to put back the logout [here](https://github.com/raycast/extensions/pull/26856/changes/b9102c45310c3fc710761bbf4b23713f366f27d9#diff-7ca6a791257e647a332d84d73c7bc7b77f13155c8fb5f9f3c626edd639f44666L775). I'm not sure if that would fix the issue, but I believe when we encounter an `Invalid session token` we should always logout.

WDYT?

File: extensions/bitwarden/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # Bitwarden Changelog
 
+## [Fixes & Improvements] - {PR_MERGE_DATE}
+
+- 🐛 Fix infinite retry loop when unlocking vault with invalid session token
+- 🐛 Surface CLI stderr in error message for invalid session token errors
+- ✨ Show unlock animation while vault is being unlocked
+- ✨ Standardize "Logged out" toast casing
```
Comment (jomifepe) [inline extensions/bitwarden/CHANGELOG.md]:
```suggestion
- 🐛 Fix infinite retry loop when unlocking vault with invalid session token
```

**Review by jomifepe — COMMENTED** [review]
Thanks for contributing! Left some comments.

File: extensions/bitwarden/src/api/bitwarden.ts [hunk 4]
```diff
@@ -423,7 +423,9 @@ export class Bitwarden {
       });
 
       const sessionToken = result.stdout;
-      if (!sessionToken.trim()) throw new InvalidSessionTokenError();
+      if (!sessionToken.trim()) {
+        throw new InvalidSessionTokenError(result.stderr?.trim() || "Invalid session token");
```
Comment (jomifepe) [inline extensions/bitwarden/src/api/bitwarden.ts]: I don't get this change 🤔 

What was the idea here?

File: extensions/bitwarden/src/components/UnlockForm.tsx (hunk 1 above)
Comment (clins1994) [reply extensions/bitwarden/src/components/UnlockForm.tsx]: thanks for the review. ok, i'll keep the pr scoped to just the login bug

File: extensions/bitwarden/src/api/bitwarden.ts (hunk 4 above)
Comment (clins1994) [reply extensions/bitwarden/src/api/bitwarden.ts]: No longer there

**Comment by clins1994** [comment]
@greptileai give this pr a fresh review please

**Comment by clins1994** [comment]
ℹ️ ran it with `npx ray dev` and tested logging in. worked fine 👍 

ℹ️ the official extension is also working for me currently so i can't really test if this fixes the edge cases from the related issues without digging deeper

File: extensions/bitwarden/src/components/UnlockForm.tsx (hunk 2 above)
Comment (jomifepe) [reply extensions/bitwarden/src/components/UnlockForm.tsx:66]: Let's not keep the PR blocked, I would suggest to put back the logout [here](https://github.com/raycast/extensions/pull/26856/changes/b9102c45310c3fc710761bbf4b23713f366f27d9#diff-7ca6a791257e647a332d84d73c7bc7b77f13155c8fb5f9f3c626edd639f44666L775).

**Review by jomifepe — APPROVED** [review]
Made one change request, but feel free to merge after!

**Review by 0xdhrv — APPROVED** [review]
Looks good to me, approved ✅
